pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -55,15 +57,24 @@ func newHorizontalReconciler(clock clock.Clock, eventRecorder record.EventRecord
 }
 
 func (hr *horizontalController) sync(ctx context.Context, podAutoscaler *datadoghq.DatadogPodAutoscaler, autoscalerInternal *model.PodAutoscalerInternal) (autoscaling.ProcessResult, error) {
+	// If we have no Spec, nothing to do
+	if autoscalerInternal.Spec() == nil {
+		return autoscaling.NoRequeue, nil
+	}
+
+	// Get the GVK of the target resource
 	gvk, err := autoscalerInternal.TargetGVK()
 	if err != nil {
+		// Resolving GVK is considered a global error, not updating horizontal last error
```

Comment:
Shouldn't we set the global error here ?

Reply:
Correct